### ISSUE TITLE
emacs-24.5 macport build fails on OS X

### ISSUE THREAD

**Issue by dhess** [NixOS/nixpkgs#12863]
Here is the failure:

```shell
Finding pointers to doc strings...
Finding pointers to doc strings...done
Dumping under the name emacs
unexec: not enough room for load commands for new __DATA segments (increase headerpad_extra in configure.in to at least 11B8)
--- List of All Regions ---
   address     size prot maxp
--- List of Regions to be Dumped ---
   address     size prot maxp
--- Header Information ---
Magic = 0xfeedfacf
CPUType = 16777223
CPUSubType = -2147483645
...
Writing unknown           command
Writing LC_FUNCTION_STARTS command
Writing LC_DATA_IN_CODE   command
Makefile:826: recipe for target 'bootstrap-emacs' failed
make[1]: *** [bootstrap-emacs] Error 1
make[1]: Leaving directory '/private/var/folders/h_/2t30cxtx7pg0b_nrzk0zdx9w0000gn/T/nix-build-emacs-24.5-mac-5.15.drv-0/emacs-24.5/src'
Makefile:393: recipe for target 'src' failed
make: *** [src] Error 2
builder for ‘/nix/store/7kn1xip1qq97s94r24lm7a1qn0gjpjv1-emacs-24.5-mac-5.15.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/k7lrxvip7nvrnqa9zxpzcnjdxwiwicfn-emacs-env.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/d40v8qjs87chpsbq61pjlc8hwnb3rsdl-exceptions-0.8.2.1.tar.gz’
cannot build derivation ‘/nix/store/8rjws9lnsnvsi5ajg3317jhll51dr4qi-hindent-4.6.1.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/vzmbd4xli7n09g8p4r3g8idgdp76g6qd-structured-haskell-mode-1.0.20.drv’: 1 dependencies couldn't be built
killing process 84549
cannot build derivation ‘/nix/store/dng0sh0514gbnwg9703y4lkngyjfzn7y-meta-env.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/dng0sh0514gbnwg9703y4lkngyjfzn7y-meta-env.drv’ failed
```

This is a common problem with Emacs on the Mac and can be fixed by adding `"-headerpad_max_install_names"` to NIX_CLFAGS_LINK. I'll submit a PR.

**Comment by matthewbauer** [comment]
This is another inactive issue I think can be closed?

**Comment by dhess** [comment]
I don't know, because the pull request I made to fix it hasn't been accepted. 

I don't know why it hasn't been accepted. The fix shouldn't be controversial as it's a very common issue when linking on OS X, and this wouldn't be the first package in Nix to use this work-around on that platform.

The discussion for the pull request went off on a tangent that had nothing to do with the fix, nor the problem that it was submitted for.

I got tired of waiting and just created my own branch which incorporates this change, and that's what I work from, so I don't know if the problem still exists in upstream Nix.
